partiql-lang/src/main/kotlin/org/partiql/lang/eval/EvaluatingCompiler.kt

```diff
@@ -150,6 +150,15 @@ internal open class EvaluatingCompiler(
             "compilationContextStack was empty.", ErrorCode.EVALUATOR_UNEXPECTED_VALUE, internal = true
         )
 
+    private val interruptionCheck: () -> Unit = when (compileOptions.interruptible) {
+        true -> { ->
+            if (Thread.interrupted()) {
+                throw InterruptedException()
+            }
+        }
+        false -> { -> }
```

Comment:
Nit: I found the false -> { -> } a little weird. Maybe: 
```suggestion
        true -> { _ -> if (Thread.interrupted()) throw InterruptedException() }
        false -> { _ -> Unit}
```